### PR TITLE
updating the readme, getting-started, how-to-contribute, and adding visualization best practices doc

### PR DESCRIPTION
**Pull request by amsaparov**
For issue #36, Margaret and I have updated the readme, getting started, and how to contribute markdown files with agreed upon changes from the google doc draft in #39.

### CODE REVIEW

File: docs/how-to-contribute.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+8. Understand the visualizations : When reviewing a PR that includes visualizations, make sure to check for the proper labeling of axes and ticks. All the values mentioned should be clear and have an explanation for each graph.
+9. Hardcoded Credentials and Paths : Sometimes developers use hardcoded credentials, secrets and even data directories for quick tests and easy access when needed and forget to remove them before publishing them to github. When reviewing a Data Science code, make sure to check for any credentials/secrets as this practice poses a significant security risk that can allow attackers to bypass authentication mechanisms and ensure that the data management is handled properly and there are no hardcoded directories in the code.
+10. Evaluate the Output : Everything looks right, but our validation results are way worse than expected. Sometimes the simplest things can go wrong like over-fitting, poorly tuned hyper-parameters or maybe an inappropriate distance metrics being used? Make sure to double-check all the smallest detail when reviewing a ML code.
+
+References :
```
Comment (hemajv) [inline docs/how-to-contribute.md:46]: can you make "References" to be as a header format?

File: docs/how-to-contribute.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+9. Once your PR is approved and merged into the repository, you can delete your branch. 
+
+## **Reviewing Pull Requests**
+
+Code reviews are common practice in the software engineering field and have been for some time. They provide developers with a well defined approach to sharing knowledge, ensuring exposure to new ideas and technologies, enforcing quality assurance of code and standards for the engineering process across both the code base and the team itself. As members of the AI CoE in the office of the CTO, we believe that having a well defined approach for how to do the same thing with data science and machine learning code is critical. Although, code is code, and we approach data science work as a subset of software engineering, there are some data science specific challenges associated with the code review process that should be addressed. Below are the 10 principles we use in our group to review data science work.
+
+1. Understand the Context : Make sure you read the pull request description and any linked issues so you know the data scientist's goal. This will help you provide an appropriate level of feedback and focus on the meaningful changes.
```
Comment (hemajv) [inline docs/how-to-contribute.md:35]: Can you bold each of the point(s) titles? i.e. make "Understand the Context" -> **Understand the Context** and similarly bold all the other point titles

**Review by hemajv — CHANGES_REQUESTED** [review]

File: docs/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+2. Create a new project description using our [template][2]. For reference, you can review an example project document [here][3].
+
+3. Put your project document into a subfolder of this [shared directory][4]. 
+
+### **Create Git Repository from a Template**
+
+Next, create a [CookieCutter][5] formatted data science project repository from this [template][6]. To do so, please follow these [instructions][7] for creating a new repository from a template.
+
+### **Set Up AICoE Continuous Integration (CI)**
+
```
Comment (oindrillac) [inline docs/getting-started.md:18]: I'm not sure how https://www.operate-first.cloud/data-science/ocp-ci-analysis/ can help with setting up the thoth bots, prow and other aspects of AICoE CI here

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Continuous Integration allows for the running of automated builds and tests of software before it is merged into a production code base. It can assist you in determining why builds are failing or certain tests are not passing.
```
Comment (oindrillac) [inline docs/getting-started.md:19]:
```suggestion
[AICoE-CI](https://github.com/aicoe/aicoe-ci#getting-started) can aid various aspects of a development workflow such as running pre-commit checks, build checks, triggering pipelines to build images etc
```

File: docs/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+For more information on AI Supported CI and how to set it up for your project, visit [this page](https://www.operate-first.cloud/data-science/ocp-ci-analysis/). Additional documentation can be found [here](https://github.com/aicoe/aicoe-ci#getting-started).
```
Comment (oindrillac) [inline docs/getting-started.md:20]:
```suggestion
For more information on AICoE CI and how to set it up for your project, you can follow the documentation [here](https://github.com/aicoe/aicoe-ci#getting-started).
```

File: docs/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+### **Create Project Boards and Issues**
+
```
Comment (oindrillac) [inline docs/getting-started.md:23]:
```suggestion
You can create Issues to highlight new features, bugs and break upcoming goals of a project into smaller chunks. You can create GitHub [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues) and include [Task Lists](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists) for each task that you plan to work on.

Here is an [example Issue](https://github.com/aicoe-aiops/ocp-ci-analysis/issues/279) that we created to highlight a certain task for our project.
```

File: docs/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+Copy this [project board][8] to a new organization level project board including automation. For instructions on how to copy a project board, review [this page][9].
```
Comment (oindrillac) [inline docs/getting-started.md:24]:
```suggestion
We use [project boards](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/managing-project-boards/about-project-boards) to organize the tasks within the team.

To get started with your project board, you can copy this [project board][8] to a new organization level project board including automation. For instructions on how to copy a project board, review [this page][9].
```

File: docs/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+Once your project board is created, begin to create issues and populate the board with them
+* Create an issue for every task you plan to work on
+    * Break issues down into smaller pieces if they cannot be completed in a single sprint
+    * Be specific in the description of the issue 
+    * Tag other related issues with a ‘#’ as necessary
+    * Provide acceptance criteria for when the issue can be closed 
+    * Assign yourself to the issue
```
Comment (oindrillac) [inline docs/getting-started.md:32]:
```suggestion
    * Assign yourself or your teammates who you plan to work with on the task to the issue
```

File: docs/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+For more information of the remainder of our GitHub workflow, please visit [this page][10]. 
+
+### **Develop on Operate First Jupyterhub**
+
+The following resources are available for use to develop your project on the Operate First JupyterHub instance:
+* View the [video tutorial](https://www.youtube.com/watch?v=iI_-lqi3vP4&list=PL8VBRDTElCWpneB4dBu4u1kHElZVWfAwW&index=3&t=13s) 
+* Access the public [JupyterHub instance][11] on the Massachusetts Open Cloud (MOC) by selecting moc-sso and then signing in with your Google Account
+* Monitor workloads using grafana dashboards (video tutorial forthcoming)
+* [Troubleshooting Runbooks](https://www.operate-first.cloud/users/sre/runbooks/jupyterhub.md)
+* Current [Grafana dashboard](https://www.google.com/url?q=https://grafana-route-opf-monitoring.apps.zero.massopen.cloud/d/24cc5f554da78f3ca60a40f190f7e23203f7d847/jupyterhub-usage?orgId%3D1%26from%3Dnow-3h%26to%3Dnow%26var-datasource%3Dopf-observatorium-thanos%26var-user_id%3Dochatter-40redhat-2ecom%26var-namespace%3Dopf-jupyterhub&sa=D&source=editors&ust=1627309997546000&usg=AOvVaw3CaL9UuS0xpAmAOwxAZCrS) for monitoring Jupyterhub
```
Comment (oindrillac) [inline docs/getting-started.md:47]:
```suggestion
* **How to Monitor JupyterHub workloads using Grafana dashboards** 

    -  (Video tutorial forthcoming)
    -  To troubleshoot issues while running your notebooks, you can refer to the [Troubleshooting Runbooks](https://www.operate-first.cloud/users/sre/runbooks/jupyterhub.md)
    -  You can refer the existing [Grafana dashboard](https://www.google.com/url?q=https://grafana-route-opf-monitoring.apps.zero.massopen.cloud/d/24cc5f554da78f3ca60a40f190f7e23203f7d847/jupyterhub-usage?orgId%3D1%26from%3Dnow-3h%26to%3Dnow%26var-datasource%3Dopf-observatorium-thanos%26var-user_id%3Dochatter-40redhat-2ecom%26var-namespace%3Dopf-jupyterhub&sa=D&source=editors&ust=1627309997546000&usg=AOvVaw3CaL9UuS0xpAmAOwxAZCrS) for monitoring your Jupyterhub workloads such as CPU, RAM usage, percentage PVC usage. 
```

File: docs/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+* When you're almost ready to begin working on an issue, move it to the *To Do* column
+* Once you start work on an issue, move it to the *In Progress* column
+
+For more information of the remainder of our GitHub workflow, please visit [this page][10]. 
+
+### **Develop on Operate First Jupyterhub**
+
+The following resources are available for use to develop your project on the Operate First JupyterHub instance:
+* View the [video tutorial](https://www.youtube.com/watch?v=iI_-lqi3vP4&list=PL8VBRDTElCWpneB4dBu4u1kHElZVWfAwW&index=3&t=13s) 
+* Access the public [JupyterHub instance][11] on the Massachusetts Open Cloud (MOC) by selecting moc-sso and then signing in with your Google Account
```
Comment (oindrillac) [inline docs/getting-started.md:44]:
```suggestion
* **Start developing on the Operate First JupyterHub environment**

    - View the [video tutorial](https://www.youtube.com/watch?v=iI_-lqi3vP4&list=PL8VBRDTElCWpneB4dBu4u1kHElZVWfAwW&index=3&t=13s) 
    - Access the public [JupyterHub instance][11] on the Massachusetts Open Cloud (MOC) by selecting moc-sso and then signing in with your Google Account
```

**Review by oindrillac — CHANGES_REQUESTED** [review]

File: docs/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+* Project Repositories ([example](https://github.com/aicoe-aiops/ocp-ci-analysis))
+* Dashboards
+* Model Services
+
+### **Creating Jupyterhub Images**
+JH images can be made from your projects to easily share reproducible ML experiments and create interactive Data Science working environments. Please refer to the following [guide](https://www.operate-first.cloud/users/data-science-workflows/docs/create_and_deploy_jh_image.md) for more information.
+
+### **Creating Automated Model Workflows**
+
+View our video [tutorial](https://www.youtube.com/watch?v=iMSOal8wRj4) for information on creating an AI Pipeline on ODH.
```
Comment (oindrillac) [inline docs/getting-started.md:70]:
```suggestion
To automate your notebook workflows using Elyra and Kubeflow Pipelines on Open Data Hub, you can refer to the [guide](https://github.com/AICoE/elyra-aidevsecops-tutorial) and view our video [tutorial](https://www.youtube.com/watch?v=iMSOal8wRj4).
```

**Review by oindrillac — COMMENTED** [review]
Thanks for putting this together @margarethaley and @amsaparov! 👏 
I have added a few suggestions 🙂

File: docs/how-to-contribute.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
+# **How to Contribute**
+
+Please refer to this guide for information on how to contribute to our data science work.
```
Comment (oindrillac) [inline docs/how-to-contribute.md:3]:
```suggestion
Please refer to this guide for information on how to contribute to our data science projects.
```

File: docs/how-to-contribute.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
```
Comment (oindrillac) [inline docs/how-to-contribute.md:4]:
```suggestion
We encourage contribution to each of our [Data Science Projects](https://www.operate-first.cloud/data-science). In order to start contributing to our projects you can check out the "How to Contribute" section for each project under https://www.operate-first.cloud/data-science/.
```

File: docs/getting-started.md (hunk 3 above)
Comment (amsaparov) [reply docs/getting-started.md:18]: Got it, would you like us to include set up for thoth bots and prow (and any other aspects) here?

Comment (oindrillac) [reply docs/getting-started.md:18]: @amsaparov let's start with linking the document. We can eventually work with the thoth team to have a detailed guide or even have an espresso series video on this! See my following suggestion

**Review by oindrillac — APPROVED** [review]
Thanks for the changes!  🚀 
/lgtm 
/approve

**Review by aakankshaduggal — APPROVED** [review]
/lgtm

**Pull request merged**